src/net_processing.cpp

```diff
@@ -3036,6 +3045,112 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, Peer& peer,
     return;
 }
 
+PeerManagerImpl::InvalidTxTask PeerManagerImpl::ProcessInvalidTx(const CTransactionRef& tx, NodeId nodeid, const TxValidationState& state)
+{
+    AssertLockHeld(cs_main);
+    AssertLockNotHeld(m_peer_mutex);
+    LogPrint(BCLog::MEMPOOLREJ, "%s (wtxid=%s) from peer=%d was not accepted: %s\n",
+             tx->GetHash().ToString(),
+             tx->GetWitnessHash().ToString(),
+             nodeid,
+             state.ToString());
+    // Maybe punish peer that gave us an tx
+    MaybePunishNodeForTx(nodeid, state);
+
+    switch (state.GetResult()) {
```

Comment:
we used to call `m_orphanage.EraseTx(orphanHash);` for each non-`TX_MISSING_INPUTS` possibility, but due to early return we don't. Can we sync this back up or justify each case this has changed for?

Reply:
Old behavior should be preserved now